src/stream/src/executor/mod.rs

```diff
@@ -457,6 +457,7 @@ impl Barrier {
             Some(
                   Mutation::Update { .. } // new actors for scaling
                 | Mutation::Add(AddMutation { pause: true, .. }) // new streaming job, or recovery
+                | Mutation::AddAndUpdate(AddMutation { pause: true, .. }, _) // new actors for replacing table
```

Comment:
In #17651 , we should add an assertion for `pause` field in `AddMutation` of `Mutation::AddAndUpdate`. Cc @shanicky